ci-operator/step-registry/cucushift/upgrade/check/capability/cucushift-upgrade-check-capability-commands.sh

```diff
@@ -0,0 +1,439 @@
+caps_operator[Console]="console"
+caps_operator[Insights]="insights"
+caps_operator[Storage]="storage"
+caps_operator[NodeTuning]="node-tuning"
+caps_operator[MachineAPI]="machine-api control-plane-machine-set cluster-autoscaler"
+caps_operator[ImageRegistry]="image-registry"
+caps_operator[OperatorLifecycleManager]="operator-lifecycle-manager operator-lifecycle-manager-catalog operator-lifecycle-manager-packageserver"
+
+# Mapping between optional capability and resources
+# Need update when new resource marks as optional
+caps_resource_list="Build DeploymentConfig"
+declare -A caps_resource
+caps_resource[Build]="builds buildconfigs"
+caps_resource[DeploymentConfig]="deploymentconfigs"
+
+# Mapping between cap sets and included capabilites
+# Need update when new cap set pops up
+declare -A caps_set
+caps_set[411]="baremetal marketplace openshift-samples"
+# shellcheck disable=SC2034
+caps_set[412]=" ${caps_set[411]} Console Insights Storage CSISnapshot"
+caps_set[413]=" ${caps_set[412]} NodeTuning"
+caps_set[414]=" ${caps_set[413]} MachineAPI Build DeploymentConfig ImageRegistry"
+caps_set[415]=" ${caps_set[414]} OperatorLifecycleManager"
+always_default="${caps_set[415]}"
+
+# Determine vCurrent
+# declare "v${source_major_version}${source_minor_version}"
+v_current_version="${target_cap_index}"
+if [[ -n ${caps_set[$v_current_version]} ]]; then
+    vCurrent=${caps_set[$v_current_version]}
+else
+    vCurrent=${always_default}
+fi
+echo "vCurrent set: $vCurrent"
+
+# Set enabled_capability_set by using baseline capset
+enabled_capability_set=""
+case ${baselinecaps_from_cluster} in
+"None")
+  ;;
+"v4.11")
+  enabled_capability_set="${caps_set[411]}"
+  (( source_minor_version >=14 && source_major_version == 4 )) && enabled_capability_set="${enabled_capability_set} MachineAPI"
+  ;;
+"v4.12")
+  enabled_capability_set="${caps_set[412]}"
+  (( source_minor_version >=14 && source_major_version == 4 )) && enabled_capability_set="${enabled_capability_set} MachineAPI"
+  ;;
+"v4.13")
+  enabled_capability_set="${caps_set[413]}"
+  (( source_minor_version >=14 && source_major_version == 4 )) && enabled_capability_set="${enabled_capability_set} MachineAPI"
+  ;;
+"v4.14")
+  enabled_capability_set="${caps_set[414]}"
+  ;;
+"v4.15")
+  enabled_capability_set="${caps_set[415]}"
+  ;;
+"vCurrent")
+  enabled_capability_set="${vCurrent}"
+  ;;
+*)
+  enabled_capability_set="${always_default}"
+  ;;
+esac
+
+# Append additional caps to enabled_capability_set
+additionalcaps_from_cluster=$(oc get clusterversion version -ojson | jq -r '.spec.capabilities.additionalEnabledCapabilities[*]')
+if [[ "${additionalcaps_from_cluster}" != "" ]]; then
+    enabled_capability_set="${enabled_capability_set} ${additionalcaps_from_cluster}"
+fi
+
+# Append new caps to enabled_capability_set
+expected_implicit_caps=""
+if ! get_new_caps "${source_cap_index}" "${target_cap_index}"; then
+    echo "ERROR: get_new_caps failed" && exit 1
```

Comment:
Shall we `exit` it as failure, if we are not very confident with that if the check is ready to move forward? Or should we always exit as passed, to check the log in early stage?

Reply:
If we `exit 0`, we will not get alert about the error so we won't know it's failed until we check the log actively. Considering the test introduced critical bugs, I'm creating a Junit for it and thinking about to merge it to `upgrade` RP attribute. So we will be notified when the step fails